tests/jax/components/executing/action_selection_test.py

```diff
@@ -0,0 +1,201 @@
+
+    Returns:
+        FeedforwardExecutorSelectAction
+    """
+    return FeedforwardExecutorSelectAction()
+
+
+@pytest.fixture
+def mock_empty_executor() -> Executor:
+    """Mock executore component with empty observations"""
+    store = SimpleNamespace(is_evaluator=None, observations={})
+    return Executor(store=store)
+
+
+def get_action(observation, rng_key, legal_actions):
+    """Function used in the networks.
+
+    Returns:
+        action_info and policy info
+    """
+    return "action_info_after_get_action_" + str(
+        observation[0]
+    ), "policy_info_after_get_action_" + str(observation[0])
+
+
+class MockExecutor(Executor):
+    def __init__(self):
+        observations = {
+            "agent_0": [0.1, 0.5, 0.7],
+            "agent_1": [0.8, 0.3, 0.7],
+            "agent_2": [0.9, 0.9, 0.8],
+        }
+        agent_net_keys = {
+            "agent_0": "network_agent_0",
+            "agent_1": "network_agent_1",
+            "agent_2": "network_agent_2",
+        }
+        networks = {
+            "networks": {
+                agent_net_keys["agent_0"]: SimpleNamespace(get_action=get_action),
+                agent_net_keys["agent_1"]: SimpleNamespace(get_action=get_action),
+                agent_net_keys["agent_2"]: SimpleNamespace(get_action=get_action),
+            }
+        }
+        key = jax.random.PRNGKey(5)
+        action_info = "action_info_test"
+        policy_info = "policy_info_test"
+
+        store = SimpleNamespace(
+            is_evaluator=None,
+            observations=observations,
+            observation=SimpleNamespace(observation=[0.1, 0.5, 0.7], legal_actions=[1]),
+            agent="agent_0",
+            networks=networks,
+            agent_net_keys=agent_net_keys,
+            key=key,
+            action_info=action_info,
+            policy_info=policy_info,
+        )
+        self.store = store
+
+    def select_action(
+        self, agent: str, observation: NestedArray, state: NestedArray = None
+    ) -> NestedArray:
+        action_info = "action_info_" + str(agent)
+        policy_info = "policy_info_" + str(agent)
+        return action_info, policy_info
+
+    def set_agent(self, agent) -> None:
+        """Update agent, observation
+
+        Args:
+            agent: the new agent to be in store.agent
+        """
+        if not agent in self.store.observations.keys():
+            pass
+
+        self.store.agent = agent
+        self.store.observation.observation = self.store.observations[agent]
+
+
+@pytest.fixture
+def mock_executor() -> Executor:
+    """Mock executor component."""
+    return MockExecutor()
+
+
+# Test initiator
+def test_constructor(dummy_config: ExecutorSelectActionConfig) -> None:
+    """Test adding config as an attribute
+
+    Args:
+        dummy_config
+    """
+    ff_executor_select_action = FeedforwardExecutorSelectAction(config=dummy_config)
+    assert ff_executor_select_action.config.parm_0 == dummy_config.parm_0
```

Comment:
In future I don't think you need to be this rigorous in testing a constructor if it doesn't do anything (this one just assigns the config), it's fine for this PR, but don't worry about it for others (unless there is some complex logic in the constructor). Reason I say this is because it can lead to confusion for someone looking at the tests as a reference for how to use the system, they could get confused as to what `parm_0` is.

Reply:
Okay, I will remove it from the next Test!